src/Listener/TerminateListener.php

```diff
@@ -165,4 +176,50 @@ private function validateInputMap(array $inputMap, string $commandClass): void
             Assert::allString($value, 'Malformed input mapper for ' . $commandClass);
         }
     }
+
+    /**
+     * @return string[]
+     */
+    private function getLocalNamespaces(): array
+    {
+        $basePath     = realpath(getcwd());
+        $autoloadFile = $basePath . '/vendor/composer/autoload_psr4.php';
+        if (! file_exists($autoloadFile)) {
+            return [];
+        }
+
+        /** @psalm-suppress MixedAssignment */
+        $namespaceMap = include $autoloadFile;
+        Assert::isMap($namespaceMap);
+
+        $localNamespaces = [];
+        $vendorDir       = $basePath . '/vendor';
```

Comment:
It might be not always `vendor`, see:
https://getcomposer.org/doc/06-config.md#vendor-dir

Reply:
Ugh, I always forget that you _can_ do this. I'll update to resolve it.

I've been thinking about your idea, and I think it's likely the more correct path forward. The main problem I have is it's going to be next-to-impossible to test that it detects 3rd party commands correctly, unless we install some as part of our dev dependencies.

Reply:
I guess we can mock it somehow, in the worst case scenario - create a file in tests in vendor 😱

Reply:
Found a way, using the "Show" command provided with Psalm. :smile: